src/libraries/System.Private.CoreLib/src/System/SpanHelpers.cs

```diff
@@ -18,11 +18,12 @@ public static unsafe void ClearWithoutReferences(ref byte b, nuint byteLength)
                 return;
 
 #if TARGET_AMD64 || TARGET_ARM64 || TARGET_LOONGARCH64
-            // The exact matrix on when ZeroMemory is faster than InitBlockUnaligned is very complex. The factors to consider include
+            // The exact matrix on when ZeroMemory is faster than SpanHelpers.Fill is very complex. The factors to consider include
             // type of hardware and memory alignment. This threshold was chosen as a good balance across different configurations.
             if (byteLength > 768)
                 goto PInvoke;
-            Unsafe.InitBlockUnaligned(ref b, 0, (uint)byteLength);
+
+            SpanHelpers.Fill(ref b, byteLength, (byte)0);
```

Comment:
I assume that the InitBlk is typically used with zero fill. It may be worth it to introduce a new JIT helper for clearing memory and use it for zero fill instead of the generic fill pattern.

Reply:
Can it be investigated & improved separately? So we can also see dotnet/performance impact from this change alone, and then from tunning if needed. As this change will require a lot of measurements on different platforms I presume..

Reply:
Ah, replied to a wrong comment

Reply:
> I assume that the InitBlk is typically used with zero fill. It may be worth it to introduce a new JIT helper for clearing memory and use it for zero fill instead of the generic fill pattern.

Agree